### PR TITLE
#P08: Resources page setup

### PR DESCRIPTION
**Pull request by irinatu17**
**User Story ID**:
#P08. As Participant, I would like to view useful resources and links.

**Description of PR**:
Adding Resources and Links page, that contains a table  with names(or descriptions) of the resources and corresponding links that open in a new tab.
- Resource **model** contains name(CharField) and link(URLField) fields.
**Note** :  the resources data can be loaded by using the following command: `python3 manage.py loaddata resources`
- Resource **view** simply renders all the entities that exist in the model: `resources = Resource.objects.all()`
- **Template**: on the medium and large screens resources' names and links are represented in the table. All the links are truncated by 50 characters: `{{ resource.link|truncatechars:50 }}`. On the smaller screens only resources' names are displayed (they are clickable, opening a link in a new tab) to achieve better UI and make it less cluttered.

**Tests**:
- Manual tests were completed and passed (clicking on all links to ensure they open in a new tab and opens corresponding pages as expected)
- Automated test were created for Resource model (checking that they can be created correctly and their string method), all passed - can be found in `tests.py` file in the resources app.
- Responsiveness testing was implemented using Google Chrome's developer tools, everything looks as expected  across all the different screen sizes.
- Browser compatibility checked on Chrome, Edge and FireFox browsers.    
  
**Know bugs/errors**:
Not found

**Screenshots**:
image1    
image2

### CODE REVIEW

File: resources/tests.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from django.test import TestCase
+from .models import Resource
+
+
+class TestResourceModels(TestCase):
+
+    def test_Resource_created_correctly(self):
+
+        resource1 = Resource.objects.create(name = 'Forking', link = "https://www.youtube.com/watch?v=HbSjyU2vf6Y")
```
Comment (stefdworschak) [inline resources/tests.py:9]: I would move creating the Resource instances in the `setUp` method?
You can then reuse it also in your 2nd test case.

File: resources/tests.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        resource2 = Resource.objects.create(name = 'The Git Story', link = "https://eventyret.github.io/the-git-story/")
+        resource1.save()
```
Comment (stefdworschak) [inline resources/tests.py:11]: Also when using `.objects.create()` you don't need to use `.save()`. The create method will automatically save it

File: resources/views.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+from django.shortcuts import render
+from .models import Resource
+
+
+def resources(request):
+    """ Display the useful resources and links page. """
+
+    template = "resources/resources.html"
+    resources = Resource.objects.all()
+    context = {
```
Comment (stefdworschak) [inline resources/views.py:10]: Not sure you need to create an extra `context` dict here just for one key. I would just pass an `{'resources': resources}` into the render function directly.

File: static/css/style.css [hunk 4]
```diff
@@ -72,3 +72,9 @@ footer {
     color: var(--s-grey);
     background-color: var(--p-grey);
 }
+
+.divider-orange{
+background-image: radial-gradient(var(--p-orange), var(--p-red));
```
Comment (stefdworschak) [inline static/css/style.css:77]: Please properly indent the CSS here.

File: resources/models.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+from django.db import models
+
+
+class Resource(models.Model):
+
```
Comment (stefdworschak) [inline resources/models.py:5]: Please add a docstring here.

File: resources/models.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+    name = models.CharField(max_length=254)
+    link = models.URLField(max_length = 200)
```
Comment (stefdworschak) [inline resources/models.py:7]: Do we only need these keys? I would suggest maybe to add a description as well to give an indication of what you would use each resource for.

**Review by stefdworschak — CHANGES_REQUESTED** [review]
@irinatu17 I really like your changes here! I left a few minor comments; if you could make the few changes, that would be great! 

Also, not sure why, but on my screen it seems that the table is a bit small. Maybe you can have a look at that as well. Here a screenshot (1920x1080):
Screenshot from 2020-10-18 00-34-09

**Comment by stefdworschak** [comment]
One last suggestion, maybe as part of this Pull Request you could also add in the resources you had in your screenshot as a json file. You can dump the data from your database by using `python3 manage.py dumpdata resources > resources.json`

**Comment by stefdworschak** [comment]
@irinatu17 just wanted to see if you have some time to look at my comments and make the suggested changes? No pressure at all if you don't have time. Just really like this PR and would be great if we could merge it 👍

File: resources/models.py (hunk 6 above)
Comment (irinatu17) [reply resources/models.py:7]: Hi @stefdworschak ! Would you suggest to make a description field required as well as name and link? Or should it be optional?

**Comment by irinatu17** [comment]
Hi @stefdworschak ! some changes are made, ready for another review!

File: resources/tests.py (hunk 1 above)
Comment (irinatu17) [reply resources/tests.py]: I'm not sure if I've done that in a correct way, never used setUp() method before. Could you have a look, please?

Comment (stefdworschak) [reply resources/tests.py]: That looks right to me!

**Review by stefdworschak — APPROVED** [review]
LGTM

@irinatu17 thanks for all the changes! This looks great!

**Pull request merged**